src/auth/googleauth.ts

```diff
@@ -268,9 +270,14 @@ export class GoogleAuth {
       this.checkIsGCE =
           res && res.headers && res.headers['metadata-flavor'] === 'Google';
     } catch (e) {
-      if ((e as NodeJS.ErrnoException).code !== 'ENOTFOUND') {
-        // Unexpected error occurred. TODO(ofrobots): retry if this was a
-        // transient error.
+      const isDNSError = (e as NodeJS.ErrnoException).code === 'ENOTFOUND';
+      const ae = e as AxiosError;
+      const is404 = ae.response && (ae.response.status === 404);
```

Comment:
Apologies. My comment wasn't clear enough. The idea is that we should retry only on 5xx errors; singling out 404 wasn't my intention.

Reply:
But we still want to not retry on ENOTFOUND, right?

Reply:
Correct.